mock/mock.spec

```diff
@@ -25,7 +25,14 @@ Requires: usermode-consoleonly
 Requires: usermode
 %endif
 Requires: createrepo_c
-Requires: mock-core-configs >= 32.4
+
+# We know that the current version of mock isn't compatible with older variants,
+# and we want to enforce automatic upgrades.
+Conflicts: mock-core-configs < 32.4
+
+# Requires 'mock-core-configs', or replacement (GitHub PR#544).
+Requires: mock-configs
```

Comment:
What do you expect to replace this with?

Reply:
There are few mock users that don't use mock-core-configs.  E.g. koji
and covscan.  They typically want to provide an alternative to this (so
why not to allow them to easily package their configs).  I expect something
like `koji-configs.rpm` which does `Provides: mock-configs`.

Reply:
Koji generates its configs on the fly, though? There are no configs on-disk for Koji.

Reply:
But they could.  Even if that was just set of few templates useful for further
koji config generator.

Reply:
TBH, I came to this topic because of covscan project;  but I could speak
at least for copr.  We keep doing manual updates of mock-core-config files all
the time, as the buildroots evolve ...  We could provide `copr-mock-configs`
for copr purposes (instead of mock-core-configs) and easily tag that into
infra-tags - without being worry that we will break anyone else.

Reply:
That said, does it make sense that someone might want to not install mock-core-configs,
and this is just compromise to allow them to do that?

Reply:
What would be different from `copr-mock-configs` from `mock-core-configs`? Generally the updates you've been making help _everyone_ using mock...

Reply:
Ideally no difference;  but practically we need sometimes to ship some change
to copr earlier.

But you are nitpicking on deployment specific problem.  Does this PR make any
sense in general?  Does it make any sense to force installation of mock-core-configs
on koji builders?

Reply:
I think there's no scenario where we would not want `mock-core-configs` installed with `mock`. The separation of the two was mainly to allow shipping updated configs without cutting a new mock release.

Reply:
What do you want `mock-core-configs` for on koji builder?

Reply:
Does it matter if it's there or not for Koji builders?

Reply:
It shouldn't.  Somewhat semantically it does, because koji could use
mock-core-configs by mistake.  It also goes against the minimization
efforts all over the place.

But I'm using this example to convince you
that mock is independent software and it _may_ have sense to have
mock installed without _default_ configuration.

Is there any other practical problem with this PR?  I mean, nothing
changes for anyone after merging this, except for those who really
want to override the configs.  Or?

Reply:
I'd add a `Suggests: mock-core-configs` to that, but yeah, otherwise it's fine.

Reply:
Frankly, covscan is pretty important customer of mock - so even if
we only helped those people this patch makes sense.  There's nothing
worse than seeing anyone re-building package you maintain with
downstream patches. :-/

Reply:
Who is `covscan`?

Reply:
s/covscan/csmock/ -> tooling for automatic static analysis (using mock):
https://github.com/kdudka/csmock

Reply:
> I'd add a Suggests: mock-core-configs to that, but yeah, otherwise it's fine.

I missed this message, ... why, when we have the Requires: there?

Reply:
You are depending on a virtual name, and having the Suggests means that `mock-core-configs` is preferred, but can be substituted by anything else, too.

Reply:
Ok, I just have to test that resolver accepts this (suggesting one package, but
installing other, conflicting package).

Reply:
I tested this.  When I did `dnf install mock test-configs`, dnf did not complain about
the Suggests at all.